drivers/pinctrl/pinctrl_lpc_iocon.c

```diff
@@ -11,22 +11,46 @@
 #include <fsl_clock.h>
 #endif
 
+/* RT7xx requests to clear the reset bit in RESET module before a IOPCTL can be used. */
+#if defined(CONFIG_SOC_SERIES_IMXRT7XX) && defined(CONFIG_SOC_FAMILY_NXP_IMXRT)
+#include <fsl_iopctl.h>
+#endif
+
+
+#if defined(CONFIG_SOC_SERIES_IMXRT7XX) && defined(CONFIG_SOC_FAMILY_NXP_IMXRT)
+/* For RT7xx , SDK driver need GPIO PORT NO and PIN NO for pin configuration purpose. */
+#define IOPCTL_PORT_NO_MASK 0xFF000000
+#define IOPCTL_PORT_NO_SHIFT 24
+#define IOPCTL_PIN_NO_MASK 0xFF0000
+#define IOPCTL_PIN_NO_SHIFT 16
+#define IOPCTL_MUX_MASK 0xFFFFUL
+#define IOPCTL_MUX_SHIFT 0
+
+#define PORT_NO(mux) (((mux) & IOPCTL_PORT_NO_MASK) >> IOPCTL_PORT_NO_SHIFT)
+#define PIN_NO(mux)  (((mux) & IOPCTL_PIN_NO_MASK) >> IOPCTL_PIN_NO_SHIFT)
+
+#else
 #define OFFSET(mux) (((mux) & 0xFFF00000) >> 20)
+static volatile uint32_t *iocon = (volatile uint32_t *)DT_REG_ADDR(DT_NODELABEL(iocon));
+#endif
+
 #define TYPE(mux) (((mux) & 0xC0000) >> 18)
 
 #define IOCON_TYPE_D 0x0
 #define IOCON_TYPE_I 0x1
 #define IOCON_TYPE_A 0x2
 
-static volatile uint32_t *iocon =
-	(volatile uint32_t *)DT_REG_ADDR(DT_NODELABEL(iocon));
-
 int pinctrl_configure_pins(const pinctrl_soc_pin_t *pins, uint8_t pin_cnt,
 			   uintptr_t reg)
 {
 	for (uint8_t i = 0; i < pin_cnt; i++) {
 		uint32_t pin_mux = pins[i];
+#if defined(CONFIG_SOC_SERIES_IMXRT7XX) && defined(CONFIG_SOC_FAMILY_NXP_IMXRT)
+		uint8_t port_no = PORT_NO(pin_mux);
```

Comment:
Why can't we use an offset here? The IOPCTL pin control registers appear continuous, just like on the RT600 and RT500

Reply:
Can't use only offset parameter to cover RT700 iopctl model because there are three iopctl instances on the RT700, and the addresses of each instance are not consecutive. The pinctrl header model has been changed to be more compatible with the SDK IOPCTL API function.

Reply:
I'd prefer to just transition the entire driver to use this macro definition if possible.

Reply:
Updated. Thank you.